File: lecture/dingcorithm/2st_week/02_12_is_palindrome_for.py
input = "abcba"


def is_palindrome(string):
    # n = len(string)
    # for i in range(n):
    #     if string[i] != string[n-1-i]:
    #         return False
    if string[0] != string[-1]:
        return False
    if len(string) <= 1:
        return True

    return is_palindrome(string[1:-1])


    # return True


print(is_palindrome(input))

File: study/dingcorithm/2st_week/02_12_is_palindrome_for.js
// 🔁 복습 필요 (3/26)(3/30)

function is_palindrome(string) {
  // 0, 4
  // 1, 3
  // 2

  // 방법 1. 제일 앞이랑 제일 뒤랑 비교해가기
  // let i;
  // for (i = 0; i <= (string.length - 1) / 2; i++) {
  //   if (string[i] !== string[string.length - 1 - i]) {
  //     return false;
  //   }
  // }

  // 방법 2. 재귀 함수 이용하기
  if (string[0] !== string[string.length - 1]) {
    return false;
  }
  if (string.length <= 1) {
    return true;
  }
  return is_palindrome(string.slice(1, -1));
}

console.log(is_palindrome("소주만병만주소"));
